### PR TITLE
[Sofa.Core] Remove the use of Link in MechanicalParams & ConstraintParams

### PR DESCRIPTION
**Pull request by damienmarchal**
Each readXX method exists in two implementation. One that takes State* while the other is taking a SingleLink<State>.

Given that from a SingleLink<State> one can get the corresponding State* using the get() method it is questionable why to maintain two set of methods for the same service.

To mimimize the surface of the API I removed the link based version and update the code base accordingly.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: Sofa/framework/Core/src/sofa/core/ConstraintParams.h [hunk 1]
```diff
@@ -121,39 +115,39 @@ class SOFA_CORE_API ConstraintParams : public sofa::core::ExecParams
 
 
 
-    /// @name Access to vectors from a given SingleLink to a state container (i.e. State or MechanicalState)
-    /// @{
+//    /// @name Access to vectors from a given SingleLink to a state container (i.e. State or MechanicalState)
```
Comment (alxbilger) [inline Sofa/framework/Core/src/sofa/core/ConstraintParams.h]: Is the plan to wait for the CI and then deprecate those functions?

Comment (damienmarchal) [reply Sofa/framework/Core/src/sofa/core/ConstraintParams.h]: For the moment I was more about first having discussion if we want to do this move. 

If everyone think this is a good move to minimize API's surface then we should go for the official deprecation-disable-remove process with this timeline:
- deprecation (from now to 22.12 release)
- disable (from 22.12 at least up to to 23.12)  as I assume the API is used in several plugin, people will have to update quickly.
- finally remove

**Comment by damienmarchal** [comment]
So do we go further (I mean setting a deprecation workflow for this one ?)

**Comment by alxbilger** [comment]
I am in favour

**Comment by damienmarchal** [comment]
I have committed the deprecation workflow. CI is compiling.

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Pull request merged**